analyze.py

```diff
@@ -374,13 +375,23 @@ def analyze(audiofile, outfile, compute_timbral_models=False, compute_descriptor
     parser.add_argument('-t', '--timbral-models', help='include descriptors computed from timbral models', action='store_const', const=True, default=False)
     parser.add_argument('-m', '--music-pieces', help='include descriptors designed for music pieces', action='store_const', const=True, default=False)
     parser.add_argument('-s', '--music-samples', help='include descriptors designed for music samples', action='store_const', const=True, default=False)
-    parser.add_argument('-i', '--input', help='input audio file', required=True)
-    parser.add_argument('-o', '--output', help='output analysis file', required=True)
+    parser.add_argument('-i', '--input', help='input audio file or input directory containing the audio files to analyze', required=True)
+    parser.add_argument('-o', '--output', help='output analysis file or output directory where the analysis files will be saved', required=True)
     parser.add_argument('-f', '--format', help='format of the output analysis file ("json" or "jsonld", defaults to "jsonld")', default="jsonld")
     parser.add_argument('-u', '--uri', help='URI for the analyzed sound (only used if "jsonld" format is chosen)', default=None)
     
     args = parser.parse_args()
     logging.basicConfig(format='%(asctime)s %(levelname)s:%(message)s', level=logging.INFO if not args.verbose else logging.DEBUG)
 
-    analyze(args.input, args.output, args.timbral_models, args.music_pieces, args.music_samples, args.format, args.uri)
-    
+    # check if the input is a file or a folder
+    if os.path.isfile(args.input):
+        analyze(args.input, args.output, args.timbral_models, args.music_pieces, args.music_samples, args.format, args.uri)
+    # ensure that the output argument is a folder
+    elif not os.path.isfile(args.output):
+        folder = args.input
+        input_files = [x for x in Path(folder).iterdir() if x.is_file()]
+        for input_file in input_files:
+            output_file = os.path.join(args.output, '{}_analysis.json'.format(input_file.stem))
+            analyze(str(input_file), output_file, args.timbral_models, args.music_pieces, args.music_samples, args.format, args.uri)
+    else:
+        Exception('Make sure input and output arguments are both files or folders')
```

Comment:
It'd be nice to make this look like a usage error from argparse, I don't know the best way to trigger this though

Reply:
I now use argparse.ArgumentTypeError.